src/steps/resource-manager/advisor/executionHandlers/recommendations.ts

```diff
@@ -0,0 +1,134 @@
+      const recommendation = getRawData<ResourceRecommendationBase>(
+        recommendationEntity,
+      )!;
+      if (recommendation.resourceMetadata?.source) {
+        const assessmentEntity = await jobState.findEntity(
+          recommendation.resourceMetadata.source,
+        );
+        if (assessmentEntity) {
+          await jobState.addRelationship(
+            createDirectRelationship({
+              _class: AdvisorRelationships.ASSESSMENT_IDENTIFIED_FINDING._class,
+              from: assessmentEntity,
+              to: recommendationEntity,
+              properties: {
+                _type: AdvisorRelationships.ASSESSMENT_IDENTIFIED_FINDING._type,
+              },
+            }),
+          );
+        }
+      }
+    },
+  );
+}
+
+export async function buildResourceToRecommendationRelationship(
+  executionContext: IntegrationStepContext,
+): Promise<void> {
+  const { jobState } = executionContext;
+
+  await jobState.iterateEntities(
+    { _type: AdvisorEntities.RECOMMENDATION._type },
+    async (recommendationEntity) => {
+      const recommendation = getRawData<ResourceRecommendationBase>(
+        recommendationEntity,
+      )!;
+      if (recommendation.resourceMetadata?.resourceId) {
+        const resourceEntity = await jobState.findEntity(
+          recommendation.resourceMetadata.resourceId,
+        );
+        if (resourceEntity) {
+          await jobState.addRelationship(
+            createDirectRelationship({
+              _class: AdvisorRelationships.ANY_RESOURCE_HAS_FINDING._class,
+              from: resourceEntity,
+              to: recommendationEntity,
+              properties: {
+                _type: AdvisorRelationships.ANY_RESOURCE_HAS_FINDING._type,
+              },
+            }),
+          );
+        }
+      }
+    },
+  );
+}
+
+export const recommendationSteps: AzureIntegrationStep[] = [
+  {
+    id: AdvisorSteps.RECOMMENDATIONS,
+    name: 'Recommendations',
+    entities: [AdvisorEntities.RECOMMENDATION],
+    relationships: [],
+    dependsOn: [
+      STEP_AD_ACCOUNT,
+      SecuritySteps.ASSESSMENTS,
+      ...getResourceManagerSteps().executeFirstSteps,
+    ],
+    executionHandler: fetchRecommendations,
+    rolePermissions: ['Microsoft.Advisor/recommendations/read'],
+  },
+  {
+    id: AdvisorSteps.ASSESSMENT_RECOMMENDATION_RELATIONSHIP,
+    name: 'Assessment to recommendation relationship',
+    entities: [],
+    relationships: [AdvisorRelationships.ASSESSMENT_IDENTIFIED_FINDING],
+    dependsOn: [AdvisorSteps.RECOMMENDATIONS],
+    executionHandler: buildAssesmentToRecommendationRelationship,
+  },
+  {
+    id: AdvisorSteps.RESOURCE_RECOMMENDATION_RELATIONSHIP,
+    name: 'Resource to recommendation relationship',
+    entities: [],
+    relationships: [AdvisorRelationships.ANY_RESOURCE_HAS_FINDING],
+    dependsOn: [AdvisorSteps.RECOMMENDATIONS],
+    executionHandler: buildResourceToRecommendationRelationship,
+    //dependencyGraphId: 'last'
```

Comment:
You'll want to replace `dependsOn` with `dependencyGraphId: 'last'` here.

```suggestion
    executionHandler: buildResourceToRecommendationRelationship,
    dependencyGraphId: 'last'
```

You'll also need to update the top-level `src/index.ts` file to include `dependencyGraphOrder`:

```diff
export const invocationConfig = {
  loadExecutionConfig,
  instanceConfigFields,
  validateInvocation,
  integrationSteps,
  getStepStartStates,
+ dependencyGraphOrder: ['last'],
};
```
See https://github.com/JupiterOne/graph-aws/blob/main/src/index.ts#L21 for an example

Reply:
Once you add `dependencyGraphId` to this step, your test will probably fail. Take a look at `graph-aws` for inspiration about how to test steps with `dependencyGraphId: 'last'`